### ISSUE TITLE
The user should be able to add icons (svg) to the list

### ISSUE THREAD

**Issue by aubreypwd** [WebDevStudios/WDS-Mega-Menu#4]
Since I [ripped out the old way](https://github.com/WebDevStudios/WDS-Mega-Menu/commit/d8a71e4efa04b5fb2cf3fb0ba288d29bb623e9ef) (using the template), we're going to need a new way to get icons in there.

We need good compatibility for SVG and custom images and maybe even callback functions. 

![](https://cloudup.com/cNMaGMZ2hIH+)

**Comment by jazzsequence** [comment]
If we allow users to add their own, they won't be in the sprite, which will increase page load time because we're loading more resources. I think that's a good feature add but I wouldn't want to change how we're defaulting to the svg sprite first.

**Comment by jazzsequence** [comment]
svgs now live in `assets/svg` and there's a grunt task to add them to `svg-defs.svg`. dunno if we want to automate a process (or if it's possible) around uploading svg files from the admin.

**Comment by aubreypwd** [comment]
Yeah, I was totally thinking that we would have default icons, but that there would be a filter or something that a _developer_ could use to add more icons. Maybe a GUI, but I wasn't thinking that direction.

**Comment by jazzsequence** [comment]
I just know that for Appirio they specifically wanted to add their own icons to our icon set that we built in their theme. And I imagine that could be a thing. Sure we can filter the hell out of the icon pack that's used and that's definitely easier but if we could figure out a way to have an actual uploader that then added those uploaded SVGs into the svg icon pack, that would be pretty sweet.
